### PR TITLE
TC-TMP-2.1: Fix

### PR DESCRIPTION
**Pull request by cecille**
The constraints here are actually relatively complex because the min and max can be null and fall back to default values.

Updated the automation to python so we can do the appropriate tests, added tests for the various scenarios becuase there are a LOT and we don't cover any of them with our current examples.

Please see
https://github.com/CHIP-Specifications/chip-test-plans/pull/4096 for test plan updates

### CODE REVIEW

File: src/python_testing/TC_TMP_2_1.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        self.step(1)
+
+        self.step(2)
+        min_bound = -27315
+        max_bound = 32767
+
+        self.step(3)
+        min_measured_value = await self.read_single_attribute_check_success(cluster=cluster, attribute=attr.MinMeasuredValue)
+        if min_measured_value != NullValue:
+            min_bound = min_measured_value
```
Comment (tehampson) [inline src/python_testing/TC_TMP_2_1.py:61]: Should we check that `min_measured_value` is not lower than current `min_bound` as that is absolute 0 already.

File: src/python_testing/TC_TMP_2_1.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+        self.step(4)
+        max_measured_value = await self.read_single_attribute_check_success(cluster=cluster, attribute=attr.MaxMeasuredValue)
+        if max_measured_value != NullValue:
+            max_bound = max_measured_value
```
Comment (tehampson) [inline src/python_testing/TC_TMP_2_1.py:66]: Similarly should we be checking that the value we get is not greater than `max_bound` which is currently max int16?

File: src/python_testing/TC_TMP_2_1.py (hunk 1 above)
Comment (tehampson) [reply src/python_testing/TC_TMP_2_1.py:61]: NVM that is step 5

File: src/python_testing/TC_TMP_2_1.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            asserts.assert_greater(max_measured_value, min_bound,
+                                   "MaxMeasuredValue is not greater than the minimum bound")
+            asserts.assert_less_equal(
+                max_measured_value, 32767, "MaxMeasuredValue is not less than or equal to than 32767")
+
+        self.step(5)
+        if min_measured_value != NullValue:
+            asserts.assert_greater_equal(min_measured_value, -27315, "")
```
Comment (tehampson) [inline src/python_testing/TC_TMP_2_1.py:74]: Consider adding error log instead of empty string

**Review by tehampson — APPROVED** [review]

File: src/python_testing/TC_TMP_2_1.py (hunk 2 above)
Comment (cecille) [reply src/python_testing/TC_TMP_2_1.py:66]: checked on line 69 below

File: src/python_testing/TC_TMP_2_1.py (hunk 3 above)
Comment (cecille) [inline src/python_testing/TC_TMP_2_1.py:74]:
```suggestion
            asserts.assert_greater_equal(min_measured_value, -27315, "MinMeasuredValue is out of range")
```

File: src/python_testing/TC_TMP_2_1.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            asserts.assert_less(min_measured_value, max_bound)
```
Comment (cecille) [inline src/python_testing/TC_TMP_2_1.py:75]:
```suggestion
            asserts.assert_less(min_measured_value, max_bound, "MinMeasuredValue is out of range")
```

File: src/python_testing/TC_TMP_2_1.py (hunk 3 above)
Comment (cecille) [reply src/python_testing/TC_TMP_2_1.py]: added.

**Pull request merged**